server/admin_management/api/app/routers/router_guards.py

```diff
@@ -121,71 +115,134 @@ async def get_target_user_from_id(
+            )
     else:
         # from body - {user_name/user_type_code}, when grant admin access
-        try:
-            rbody = await request.json()
-            user = user_service.get_user_by_domain_and_name(
-                rbody["user_type_code"],
-                rbody["user_name"],
-            )
-            return TargetUser.model_validate(user) if user is not None else None
-        except json.JSONDecodeError:
-            return None
+        rbody = await request.json()
+        LOGGER.debug(f"Dependency 'get_target_user_from_id' request body {rbody}.")
+        rb_user_name = rbody["user_name"]
+        rb_user_type_code = rbody["user_type_code"]
+        # user_guid is searched from IDIM search and passed from frontend.
+        rb_user_guid = rbody["user_guid"]
+
+        user = user_service.get_user_by_domain_and_name(
+            rb_user_type_code,
+            rb_user_name,
+        )
+        if user is not None:
+            found_target_user = TargetUser.model_validate(user)
+            # old data might not have user_guid.
+            if found_target_user.user_guid is None:
+                found_target_user.user_guid = rb_user_guid
+            return found_target_user
+        # user not found, case of new user.
+        else:
+            target_new_user = TargetUser.model_validate({
+                "user_name": rb_user_name,
+                "user_type_code": rb_user_type_code,
+                "user_guid": rb_user_guid
+            })
+            return target_new_user
+
+
+async def target_user_bceid_search(
+    requester: Requester = Depends(get_current_requester),
+    target_user: TargetUser = Depends(get_target_user_from_id)
+) -> TargetUser:
+    """
+    BCeID user search for target_user.
+    :param target_user "Depends(get_target_user_from_id)": the initial
+        dependency with "get_target_user_from_id" has initial target_user
+        parsed from the request. It's business_guid will be updated to the
+        target_user after user is searched through IDIM Proxy.
+    """
+    if (
+        target_user.user_type_code == UserType.BCEID and
+        (
+            target_user.is_new_user() or
+            target_user.business_guid is None
+        )
+    ):
+        user_guid = target_user.user_guid  # will be used by search using user_guid
+        LOGGER.debug(
+            f"Searching business guid for target_user: {target_user}, with requester {requester}"
+        )
+        idim_proxy_api = IdimProxyService(requester)
+        search_result = idim_proxy_api.search_business_bceid(
+            IdimProxyBceidSearchParam(**{
+                "searchUserBy": IdimSearchUserParamType.USER_GUID,
+                "searchValue": user_guid
+            })
+        )
+        business_guid = search_result.get("businessGuid")
+        LOGGER.debug(
+            f"business_guid result: {business_guid} is updated to target_user."
+        )
+        target_user.business_guid = business_guid
```

Comment:
Update business_guid after search for target_user. And return updated target_user for router handler.